community/kernel/src/main/java/org/neo4j/kernel/impl/api/LookupFilter.java

```diff
@@ -96,17 +96,10 @@ private static boolean isNumericOrGeometricPredicate( IndexQuery predicate )
         if ( predicate.type() == IndexQuery.IndexQueryType.exact )
         {
             IndexQuery.ExactPredicate exactPredicate = (IndexQuery.ExactPredicate) predicate;
-            if ( isNumberGeometryOrArray( exactPredicate.value() ) )
-            {
-                return true;
-            }
-        }
-        else if ( predicate.type() == IndexQuery.IndexQueryType.range &&
-                  ( predicate.valueGroup() == ValueGroup.NUMBER || predicate.valueGroup() == ValueGroup.GEOMETRY ) )
-        {
-            return true;
+            return isNumberGeometryOrArray(exactPredicate.value());
         }
-        return false;
+        else return predicate.type() == IndexQuery.IndexQueryType.range &&
+                (predicate.valueGroup() == ValueGroup.NUMBER || predicate.valueGroup() == ValueGroup.GEOMETRY);
```

Comment:
I think this is harder to read, because we previously searched for specific types and otherwise fell through to return `false` by default. Now, the fall through is folded into a more complicated expression.